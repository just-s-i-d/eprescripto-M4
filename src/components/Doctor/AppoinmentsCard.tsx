import { useState } from "react";
import { Card, Skeleton, Avatar, Tooltip, Image } from "antd";
import { CloseCircleTwoTone, PhoneTwoTone } from "@ant-design/icons";

import noDataPng from "@assets/no-data.png";
import CardTitle from "@components/ui/CardTitle";
import PopModal from "@components/ui/PopModal";
import profilePic from "@assets/profilepic.png";
import {
  ApiResponseData,
  ApiResponseDataType,
  AppointmentDataType,
} from "@constants/types";
import { cancelAppointment } from "@utils/Doctor";
import ErrorBoundary from "@components/ErrorBoundary";
import useStatesHook from "src/hooks/useStatesHook";
import { showToast } from "@utils/common";

type AppointmentCardPropsType = {
  appointments?: ApiResponseData<AppointmentDataType>;
  loading: boolean;
  appointmentsState: ReturnType<
    typeof useStatesHook<ApiResponseData<AppointmentDataType>>
  >;
  getDataForAppointments: () => void;
};

const AppoinmentsCard = ({
  appointments,
  loading,
  appointmentsState,
  getDataForAppointments,
}: AppointmentCardPropsType) => {
  const [open, setOpen] = useState<boolean>(false);
  const [selectedPatient, setSelectedPatient] =
    useState<ApiResponseDataType<AppointmentDataType>>();
  const handlerCancleAppointment = (
    patient: ApiResponseDataType<AppointmentDataType>,
  ) => {
    setOpen(true);
    setSelectedPatient(patient);
  };
  const confirmHandler = async () => {
    if (selectedPatient) {
      cancelAppointment(selectedPatient)
        .then((res) => {
          showToast(res.message, res.type);
          getDataForAppointments();
        })
        .catch((error) => {
          showToast(error.message, error.type);
        });
    }
  };
  return (
    <>
      <Card
        className={`w-[32%] max-xxl:h-[59vh] max-h-[59vh] max-xl:max-h-[60vh] max-xxl:w-[48%] max-xl:w-[48%] max-md:w-9/12 max-sm:w-full `}
        bordered={false}
      >
        <ErrorBoundary
          refreshComponent={() => appointmentsState.setRefresh((prev) => !prev)}
          error={appointmentsState.error}
        >
          <CardTitle>Today's Appointments</CardTitle>
          {!appointmentsState.loading && !appointmentsState.data?.length ? (
            <div className="flex flex-col items-center justify-center">
              <Image src={noDataPng} preview={false} />
              <h2 className="text-lg mt-12 text-center">No Appointments</h2>
            </div>
          ) : (
            <div className="mx-auto px-4 h-[44vh] overflow-y-auto">
              <ul className="mt-1 divide-y">
                {appointments?.map((item, idx) => (
                  <li
                    key={idx}
                    className="py-4 mb-2 px-2 bg-[#F0F3F4] rounded-lg"
                  >
                    <div className="flex items-start justify-between">
                      <div className="flex gap-3">
                        {loading ? (
                          <Skeleton.Avatar active size={48} />
                        ) : (
                          <Avatar
                            size={48}
                            src={item.attributes?.profilePic || profilePic}
                          />
                        )}
                        <Skeleton
                          loading={loading}
                          active
                          avatar={false}
                          title={false}
                          paragraph={{ rows: 2, width: [100, 100] }}
                        >
                          <div>
                            <span className="block text-base text-gray-700">
                              {item.attributes?.pName}
                            </span>
                            <span className="block text-[12px] text-gray-600">
                              {item.attributes?.timeSlot}
                            </span>
                          </div>
                        </Skeleton>
                      </div>
                      {loading ? (
                        <Skeleton.Button active />
                      ) : (
                        <span
                          className={`py-1 px-2 rounded-lg ${item.attributes?.status.toLowerCase()}`}
                        >
                          {item.attributes?.status}
                        </span>
                      )}
                    </div>
                    <div className="flex justify-between">
                      <span className="block text-sm text-gray-700 font-bold">
                        <span className="mr-1 py-1 text-gray-500 font-semibold">
                          Referrer
                        </span>
                        {item.attributes.referrer || "NA"}
                      </span>
                      <div className="flex justify-end gap-4 text-lg">
                        <Tooltip title={item.attributes?.contact}>
                          <PhoneTwoTone
                            className="rotate-90 cursor-pointer"
                            href={`tel:${item.attributes?.contact}`}
                          />
                        </Tooltip>

                        {item.attributes.status === "Pending" && (
                          <CloseCircleTwoTone
                            className="cursor-pointer"
                            onClick={() => handlerCancleAppointment(item)}
                          />
                        )}
                      </div>
                      <PopModal
                        setOpen={setOpen}
                        open={open}
                        okButtonText="Sure"
                        title="Cancel Appointment"
                        footer={true}
                        confirmHandler={confirmHandler}
                      >
                        Are you sure,you want to cancel the appoinment for{" "}
                        {selectedPatient?.attributes.pName}?
                      </PopModal>
                    </div>
                  </li>
                ))}
              </ul>
            </div>
          )}
        </ErrorBoundary>
      </Card>
    </>
  );
};

export default AppoinmentsCard;
