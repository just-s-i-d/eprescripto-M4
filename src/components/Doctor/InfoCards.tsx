import { useEffect, useState } from "react";
import { Row, Col, Card, Image, Typography, Skeleton } from "antd";

import ErrorBoundary from "@components/ErrorBoundary";
import CardTitle from "@components/ui/CardTitle";
import { ApiResponseData, InfoCardDetailsType } from "@constants/types";
import { CARD_PROPERTIES } from "@constants/constants";
import useStatesHook from "../../hooks/useStatesHook";

const { Text } = Typography;

type InfoCardsProps = {
  cardDetails?: ApiResponseData<InfoCardDetailsType>;
  infoCards: ReturnType<
    typeof useStatesHook<ApiResponseData<InfoCardDetailsType>>
  >;
};

const getBgColor = (index: number) => {
  switch (index + 1) {
    case 1:
      return "bg-infoCardColors-color1";
    case 2:
      return "bg-infoCardColors-color2";
    case 3:
      return "bg-infoCardColors-color3";
    case 4:
      return "bg-infoCardColors-color4";
  }
};
const InfoCards = ({ cardDetails, infoCards }: InfoCardsProps) => {
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    cardDetails && setTimeout(() => setLoading(false), 500);
  }, [cardDetails]);
  return (
    <Card
      className="info-cards-body w-full h-[22vh] max-md:h-full flex justify-center"
      bordered={false}
    >
      <ErrorBoundary
        error={infoCards.error}
        refreshComponent={() => infoCards.setRefresh((prev) => !prev)}
      >
        <Row
          gutter={30}
          className="w-full max-md:flex max-md:flex-wrap gap-y-6 justify-center max-lg:flex-nowrap"
        >
          {cardDetails?.map((element, index) => {
            return (
              <ErrorBoundary key={element.attributes.title}>
                <Col className="basis-3/12 max-md:basis-6/12 max-sm:basis-9/12">
                  <Card
                    className={`info-card h-[17vh] py-8 px-4 max-sm:py-5 max-w-[390px] max-xl:h-[16vh] max-xl:min-w-[135px] max-sm:h-[150px] ${getBgColor(
                      index,
                    )} text-white`}
                    bordered={false}
                  >
                    <div>
                      <div className={`flex items-center gap-4 mb-2 mt-2`}>
                        {loading ? (
                          <Skeleton.Avatar active shape="square" size={50} />
                        ) : (
                          <Image
                            src={CARD_PROPERTIES[index]}
                            alt="Icon image"
                            className="w-[50px] max-xxl:w-[35px] max-md:w-[40px]"
                            preview={false}
                          />
                        )}
                        {loading ? (
                          <Skeleton.Button />
                        ) : (
                          <Text className="text-4xl max-xl:text-3xl text-white max-md:text-4xl">
                            {element.attributes.totalCount}
                          </Text>
                        )}
                      </div>
                      {loading ? (
                        <Skeleton.Input active />
                      ) : (
                        <CardTitle className="max-xxl:text-lg max-xl:text-base max-md:text-lg">
                          {element.attributes.title}
                        </CardTitle>
                      )}
                    </div>
                  </Card>
                </Col>
              </ErrorBoundary>
            );
          })}
        </Row>
      </ErrorBoundary>
    </Card>
  );
};
export default InfoCards;
