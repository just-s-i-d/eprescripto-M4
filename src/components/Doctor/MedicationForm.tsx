import { Input, Select, Form } from "antd";
import { CloseOutlined } from "@ant-design/icons";
import { MedicationType } from "@constants/types";

const { Item } = Form;
type MedicationFormPropsType = {
  fields: MedicationType;
  removeable?: boolean;
  removeMedicationInput: (id: number) => void;
};
const MedicationForm = ({
  fields,
  removeMedicationInput,
  removeable,
}: MedicationFormPropsType) => {
  return (
    <>
      <div className="mt-4 mb-2 flex justify-between">
        <h2 className="text-xl">Medication</h2>
        {removeable && (
          <CloseOutlined
            onClick={() => removeMedicationInput(fields.id)}
            data-testid="remove-field"
            className="text-[16px]"
          />
        )}
      </div>

      <div className="w-full grid sm:grid-cols-2 md:grid-cols-4 gap-x-5 gap-y-3">
        <Item
          name={fields.medicine}
          label="Medicine Name"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[a-zA-Z0-9.-]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.dosage}
          label="Dosage"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[0-9/mg]/,
              message: "Enter a valid dosage",
            },
          ]}
        >
          <Input />
        </Item>
        <Item
          name={fields.timesPerDay}
          label="Frequency"
          className="w-full"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
          ]}
        >
          <Select
            options={[
              {
                value: 1,
                label: "1/Day",
              },
              {
                value: 2,
                label: "2/Day",
              },
              {
                value: 3,
                label: "3/Day",
              },
            ]}
          />
        </Item>
        <div className="flex gap-6">
          <Item
            name={fields.instruction}
            className="w-full"
            label="Instruction"
            required
            rules={[
              {
                required: true,
                message: "Field cannot be empty",
              },
            ]}
          >
            <Select
              options={[
                {
                  value: "beforeFood",
                  label: "Before eating food",
                },
                {
                  value: "afterFood",
                  label: "After eating Food",
                },
              ]}
            />
          </Item>
        </div>
      </div>
    </>
  );
};
export default MedicationForm;
