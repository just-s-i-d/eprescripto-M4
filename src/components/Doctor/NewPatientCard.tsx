import ErrorBoundary from "@components/ErrorBoundary";
import { Form, FormInstance, Input, Select } from "antd";
import FormItem from "antd/es/form/FormItem";
import TextArea from "antd/es/input/TextArea";

type NewPatientCardPropType = {
  newPatientForm: FormInstance;
};
const NewPatientCard = ({ newPatientForm }: NewPatientCardPropType) => {
  return (
    <ErrorBoundary>
      <Form form={newPatientForm} layout="vertical" className="mt-6 px-4">
        <FormItem
          name="pName"
          label="Patient Name"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[A-Za-z\s]+$/,
              message: "Special characters not allowed",
            },
          ]}
        >
          <Input />
        </FormItem>
        <FormItem
          name="email"
          label="Email"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[\w.-]+@[\w.-]+\.\w+$/,
              message: "Enter a valid mail",
            },
          ]}
        >
          <Input />
        </FormItem>
        <div className="flex gap-6">
          <FormItem
            name="dob"
            label="Date of Birth"
            required
            className="w-full"
            rules={[
              {
                required: true,
                message: "Please enter your age!",
              },
            ]}
          >
            <Input type="date" />
          </FormItem>
          <FormItem
            name="gender"
            label="Gender"
            required
            className="w-full"
            rules={[
              {
                required: true,
                message: "Please select your gender!",
              },
            ]}
          >
            <Select
              placeholder="Select your gender"
              options={[
                { value: "male", label: "Male" },
                { value: "female", label: "Female" },
                { value: "other", label: "Other" },
              ]}
            />
          </FormItem>
        </div>
        <FormItem
          name="contactNo"
          label="Mobile"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              pattern: /^[6-9]\d{9}$/,
              message: "Enter valid no.",
            },
          ]}
        >
          <Input />
        </FormItem>
        <FormItem name="allergies" label="Mention allergies , if any">
          <TextArea placeholder="Mention your allergies here" />
        </FormItem>
      </Form>
    </ErrorBoundary>
  );
};
export default NewPatientCard;
