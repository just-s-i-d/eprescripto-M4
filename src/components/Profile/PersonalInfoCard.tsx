import CardTitle from "@components/ui/CardTitle";
import { Button, Card, Form, Input, Select, Space } from "antd";
import profilePic from "@assets/profilepic.png";
import { useState } from "react";
const PersonalInfoCard = () => {
  const [disabled, setDisabled] = useState(true);
  return (
    <Card>
      <CardTitle>Personal Information</CardTitle>
      <div className="flex justify-between">
        <div>
          <Form className="flex flex-col">
            <Space>
              <Form.Item
                name="docName"
                label="Full Name"
                rules={[
                  {
                    required: true,
                    message: "Please enter your name!",
                  },
                ]}
              >
                <Input disabled={disabled} />
              </Form.Item>
              <Form.Item
                name="email"
                label="E-mail"
                rules={[
                  {
                    required: true,
                    message: "Please enter your email!",
                  },
                ]}
              >
                <Input disabled={disabled} />
              </Form.Item>
            </Space>
            <Space>
              <Form.Item
                name="age"
                label="Age"
                rules={[
                  {
                    required: true,
                    message: "Please enter your name!",
                  },
                ]}
              >
                <Input disabled={disabled} />
              </Form.Item>
              <Form.Item
                name="gender"
                label="Gender"
                rules={[
                  {
                    required: true,
                    message: "Please enter your email!",
                  },
                ]}
              >
                <Select
                  defaultValue="default"
                  className="w-[200px]"
                  options={[
                    { value: "default", label: "Select a gender", disabled },
                    { value: "male", label: "Male" },
                    { value: "female", label: "Female" },
                    { value: "other", label: "Other" },
                  ]}
                />
              </Form.Item>
            </Space>
            <Space>
              <Form.Item
                name="city"
                label="City"
                rules={[
                  {
                    required: true,
                    message: "Please enter your City!",
                  },
                ]}
              >
                <Input disabled={disabled} />
              </Form.Item>
              <Form.Item
                name="state"
                label="State"
                rules={[
                  {
                    required: true,
                    message: "Please enter your State!",
                  },
                ]}
              >
                <Input disabled={disabled} />
              </Form.Item>
            </Space>
            <Space>
              {disabled ? (
                <>
                  <Button onClick={() => setDisabled(false)}>
                    Edit Details
                  </Button>
                  <Button>Change Password</Button>
                  <Button danger>Delete Account</Button>
                </>
              ) : (
                <>
                  <Button>Save</Button>
                  <Button onClick={() => setDisabled(true)}>Cancel</Button>
                </>
              )}
            </Space>
          </Form>
        </div>
        <div className="flex flex-col justify-center gap-6">
          <img src={profilePic} alt="" width={150} />
          <Button>Change Picture</Button>
        </div>
      </div>
    </Card>
  );
};
export default PersonalInfoCard;
