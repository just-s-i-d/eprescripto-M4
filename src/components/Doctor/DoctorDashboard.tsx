import { useEffect, useState } from "react";

import NextPatientCard from "./NextPatientCard";
import InfoCards from "./InfoCards";
import ErrorBoundary from "@components/ErrorBoundary";
import PieChart from "../Charts/PieChart";
import LineChart from "@components/Charts/LineChart";
import BarChart from "@components/Charts/BarChart";
import AppoinmentsCard from "./AppoinmentsCard";
import {
  AppointmentsDataType,
  ChartDataType,
  InfoCardDetailsType,
  LineChartApiResDataType,
} from "@constants/types";
import {
  barChartDataEndPoint,
  getAppointmentData,
  getChartData,
  getDataForLineGraph,
  getInfoCardsData,
  lineChartDataEndPoint,
  pieChartDataEndPoint,
} from "@utils/Doctor";

const DoctorDashboard: React.FC = () => {
  const [lineChartData, setLineChartData] = useState<LineChartApiResDataType>();
  const [infoCardError, setInfoCardError] = useState(false);
  const [appointmentCardError, setAppointmentCardError] = useState(false);
  const [appointments, setAppointments] = useState<AppointmentsDataType>();
  const [infoCardsData, setInfoCardsData] = useState<InfoCardDetailsType>();
  const [pieChartData, setPieChartData] = useState<ChartDataType>();
  const [barChartData, setBarChartData] = useState<ChartDataType>();

  useEffect(() => {
    getInfoCardsData()
      .then((res) => {
        setInfoCardsData(res);
        setInfoCardError(false);
      })
      .catch(() => setInfoCardError(true));
    getAppointmentData()
      .then((res) => setAppointments(res))
      .catch(() => setAppointmentCardError(true));
    getChartData(pieChartDataEndPoint).then((res) => {
      setPieChartData(res.pieChartData);
    });
    getChartData(lineChartDataEndPoint).then((res) => {
      setLineChartData(res.lineChartData);
    });
    getChartData(barChartDataEndPoint).then((res) => {
      setBarChartData(res.barChartData);
    });
  }, []);
  return (
    <>
      <ErrorBoundary>
        <div className="max-w-full flex">
          <div className="w-full flex flex-col gap-6">
            <ErrorBoundary error={infoCardError}>
              <InfoCards cardDetails={infoCardsData} error={infoCardError} />
            </ErrorBoundary>
            <div className="flex flex-wrap justify-between gap-y-6 max-tablet:justify-center">
              <ErrorBoundary>
                <NextPatientCard
                  appointment={appointments && appointments[0]}
                  error={appointmentCardError}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <PieChart
                  chartData={pieChartData}
                  title="Ratings Composition"
                  label="Reviews"
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <AppoinmentsCard
                  appointments={appointments}
                  loading={appointmentCardError}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <LineChart
                  chartData={lineChartData}
                  title="Patients in Last 7 Days"
                  xAxesTitle="Dates"
                  yAxesTitle="Number of Patients"
                  getDataForLineGraph={getDataForLineGraph}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <BarChart
                  chartData={barChartData}
                  title="Age Groups Comparison"
                  label="Patients"
                  xAxesTitle="Different Age Groups (In Years)"
                  yAxesTitle="Number of Patients"
                />
              </ErrorBoundary>
            </div>
          </div>
        </div>
      </ErrorBoundary>
    </>
  );
};

export default DoctorDashboard;
