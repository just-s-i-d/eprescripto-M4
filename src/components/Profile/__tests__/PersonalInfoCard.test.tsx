import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@test/matchMedia";
import PersonalInfoCard from "../PersonalInfoCard";
import { updateUserData } from "@utils/Doctor";

jest.mock("@utils/Doctor", () => ({
  updateUserData: jest.fn().mockResolvedValue("updated"),
}));
const mockUserData = {
  docName: "John Doe",
  email: "john.doe@example.com",
  dob: "1/11/1999",
  gender: "male",
  city: "City",
  state: "State",
  country: "Country",
  profilePic: "profilePic.jpg",
};

describe("PersonalInfoCard", () => {
  it("Render the card with given user details", async () => {
    const { asFragment } = render(<PersonalInfoCard userData={mockUserData} />);
    expect(screen.getByText("Personal Information")).toBeInTheDocument();
    expect(screen.getByLabelText("Full Name")).toHaveValue(
      mockUserData.docName,
    );
    expect(screen.getByLabelText("E-mail")).toHaveValue(mockUserData.email);
    expect(screen.getByLabelText("City")).toHaveValue(mockUserData.city);
    expect(screen.getByLabelText("State")).toHaveValue(mockUserData.state);
    expect(screen.getByLabelText("Country")).toHaveValue(mockUserData.country);
    expect(asFragment()).toMatchSnapshot();
  });

  it("allows editing and saving user details", async () => {
    render(<PersonalInfoCard userData={mockUserData} />);
    fireEvent.click(screen.getByText("Edit Details"));
    fireEvent.change(screen.getByLabelText("Full Name"), {
      target: { value: "New Name" },
    });
    fireEvent.click(screen.getByText("Save"));
    expect(updateUserData).toHaveBeenCalledWith({
      ...mockUserData,
      docName: "New Name",
    });
  });
  it("User details update error", async () => {
    (updateUserData as jest.Mock).mockRejectedValue("mocked");
    render(<PersonalInfoCard userData={mockUserData} />);
    const editDetailsBtn = screen.getByText("Edit Details");
    fireEvent.click(editDetailsBtn);
    fireEvent.change(screen.getByLabelText("Full Name"), {
      target: { value: "New Name" },
    });
    const saveBtn = screen.getByText("Save");
    fireEvent.click(saveBtn);
    expect(updateUserData).toHaveBeenCalledWith({
      ...mockUserData,
      docName: "New Name",
    });
    await waitFor(() => {
      expect(screen.getByText("Edit Details")).toBeInTheDocument();
      expect(screen.queryByText("Save")).not.toBeInTheDocument();
    });
  });
  it("cancels the editing of user details", async () => {
    render(<PersonalInfoCard userData={mockUserData} />);
    fireEvent.click(screen.getByText("Edit Details"));
    fireEvent.change(screen.getByLabelText("Full Name"), {
      target: { value: "New Name" },
    });
    fireEvent.click(screen.getByText("Cancel"));
    expect(screen.getByText("Edit Details")).toBeInTheDocument();
    expect(screen.getByLabelText("Full Name")).toHaveValue(
      mockUserData.docName,
    );
    expect(screen.getByLabelText("City")).toHaveValue(mockUserData.city);
  });
});
