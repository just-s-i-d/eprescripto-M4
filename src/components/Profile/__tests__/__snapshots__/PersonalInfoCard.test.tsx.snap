// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfoCard Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-card-body"
    >
      <h1
        class="text-xl font-bold mb-6 undefined"
      >
        Personal Information
      </h1>
      <div>
        <form
          class="ant-form ant-form-vertical css-dev-only-do-not-override-1adbn6x custom-form custom-form-profile w-full flex xs:flex-col xs:items-start sm:items-center sm:justify-between mb-4  sm:flex-row-reverse"
        >
          <div
            class="flex justify-center xs:w-full sm:w-4/12"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        accept="image/*"
                        class="ant-input css-dev-only-do-not-override-1adbn6x hidden"
                        id="imageInput"
                        type="file"
                        value=""
                      />
                      <span
                        class="flex flex-col items-center justify-center"
                      >
                        <img
                          alt="Profile Preview"
                          class="xs:w-[120px] md:w-[150px] rounded-[50%] mb-4"
                          src="profilePic.jpg"
                        />
                        <button
                          class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default w-full"
                          disabled=""
                          type="button"
                        >
                          <label
                            for="imageInput"
                          >
                            Change Picture
                          </label>
                        </button>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-3 xs:w-full sm:w-7/12 md:w-9/12 xl:w-8/12"
          >
            <div
              class="flex xs:gap-3 md:gap-6 xs:flex-col md:flex-row"
            >
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="docName"
                      title="Full Name"
                    >
                      Full Name
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="docName"
                          type="text"
                          value="John Doe"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="email"
                      title="E-mail"
                    >
                      E-mail
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="email"
                          type="text"
                          value="john.doe@example.com"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex xs:gap-3 md:gap-6 xs:flex-col md:flex-row"
            >
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="dob"
                      title="Date of birth"
                    >
                      Date of birth
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="dob"
                          type="date"
                          value="1/11/1999"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="gender"
                      title="Gender"
                    >
                      Gender
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          aria-required="true"
                          class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow ant-select-disabled"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-autocomplete="list"
                                aria-controls="gender_list"
                                aria-expanded="false"
                                aria-haspopup="listbox"
                                aria-owns="gender_list"
                                aria-required="true"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                disabled=""
                                id="gender"
                                readonly=""
                                role="combobox"
                                style="opacity: 0;"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-item"
                              title="Male"
                            >
                              Male
                            </span>
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select: none;"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex xs:gap-3 md:gap-6 xs:flex-col md:flex-row"
            >
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="city"
                      title="City"
                    >
                      City
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="city"
                          type="text"
                          value="City"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="state"
                      title="State"
                    >
                      State
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="state"
                          type="text"
                          value="State"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item xs:w-full md:w-[48%] xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                >
                  <label
                    class="ant-form-item-required"
                    for="country"
                    title="Country"
                  >
                    Country
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                        disabled=""
                        id="country"
                        type="text"
                        value="Country"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="flex xs:gap-3 md:gap-6"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
            type="button"
          >
            <span>
              Edit Details
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
